presto-docs/src/main/sphinx/connector/mongodb.rst

```diff
@@ -0,0 +1,242 @@
+
+#. Connect in replica set mode, and discover all members of the set based on the given servers
+#. Make sure that the set name reported by all members matches the required set name.
+#. Refuse to service any requests if any member of the seed list is not part of a replica set with the required name.
+
+This property is optional; no default value
+
+``mongodb.required-replica-set``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Limits the number of elements returned in one batch. A cursor typically fetches a batch of result objects and store them locally.
+If batchSize is 0, Driver's default will be used.
+If batchSize is positive, it represents the size of each batch of objects retrieved. It can be adjusted to optimize performance and limit data transfer.
+If batchSize is negative, it will limit of number objects returned, that fit within the max batch size limit (usually 4MB), and cursor will be closed. For example if batchSize is -10, then the server will return a maximum of 10 documents and as many as can fit in 4MB, then close the cursor.
+
+.. note::
+    Do not use a batch size of 1.
+
+This property is optional; the default is ``0``.
+
+Table Definition
+----------------
+
+MongoDB maintains table definitions on the special collection where ``mongodb.schema-collection`` configuration value specifies.
+
+.. note::
+    There's no way for the plugin to detect a collection is deleted.
+    You need to delete the entry by ``db.getCollection("_schema").remove( { table: deleted_table_name })``. Or please drop a collection by ``drop table table_name`` at Presto shell.
+
+A schema collection consists of a MongoDB document for a table.
+
+.. code-block:: json
+
+    {
+        "table": ...,
+        "fields": [
+              { "name" : ...,
+                "type" : "varchar|bigint|boolean|double|date|array<bigint>|...",
+                "hidden" : false },
+                ...
+            ]
+        }
+    }
+
+=============== ========= ============== =============================
+Field           Required  Type           Description
+=============== ========= ============== =============================
+``table``       required  string         Presto table name
+``fields``      required  array          A list of field definitions. Each field definition creates a new column in the Presto table.
+=============== ========= ============== =============================
+
+Each field definition:
+
+.. code-block:: json
+
+    {
+        "name": ...,
+        "type": ...,
+        "hidden": ...
+    }
+
+=============== ========= ========= =============================
+Field           Required  Type      Description
+=============== ========= ========= =============================
+``name``        required  string    Name of the column in the Presto table.
+``type``        required  string    Presto type of the column.
+``hidden``      optional  boolean   Hides the column from ``DESCRIBE <table name>`` and ``SELECT *``. Defaults to ``false``.
```

Comment:
Basically, users might not want to expose some columns of a table. I guess why Presto has `hidden` property in the `ColumnMetadata`